datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +41,22 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
+
+    Map<String, PartitionThroughputInfo> throughputMap = throughputInfo.getPartitionInfoMap();
```

Comment:
What happens in the throughput info is not available ? Will it be the callers responsibility in that case? It will be better to add those information.

Reply:
I was thinking about this too. What if throughput info is not available for a subset of partitions? I think it makes sense to just print a warning message in that case. Worst case scenario, the old mechanism for task count estimation will win, but a warning message would be helpful to have.

Reply:
There's a slight problem here though. If we want to filter the provider results and get only info for partitions with throughput > X, there won't be a way to tell if the throughput info was filtered out or it wasn't available.

Reply:
Empty map can be treated as throughput not available. In case of Pinot, you will get success or failure. In case of Brooklin monitor, you can assume and retry.

Reply:
Also, Throughput > X is very critical to get the results faster in case of Pinot.